packages/web-extension/src/messaging/types.ts

```diff
@@ -159,11 +159,17 @@ export interface DeriveChannelOptions {
   detached?: boolean;
 }
 
+export interface DisconnectEvent {
+  disconnected: MinimalPort;
+  remaining: MinimalPort[];
+}
+
 export interface Messenger extends Shutdown {
   channel: ChannelName;
   connect$: Observable<MinimalPort>;
   postMessage(message: unknown): Observable<void>;
   message$: Observable<PortMessage>;
+  disconnect$: Observable<DisconnectEvent>;
```

Comment:
When squashing, please mark the commit as breaking, as other possible implementations of the interface will be broken, also worth pointing it out in the commit body notes.

Reply:
Yeah, I missed this detail